src/node/blockstorage.h

```diff
@@ -156,10 +156,15 @@ class BlockManager
     [[nodiscard]] bool FlushUndoFile(int block_file, bool finalize = false);
 
     /**
+     * Helper function performing various preparations before a block can be saved to disk:
+     * Returns the correct position for the block to be saved, which may be in the current or a new
+     * block file depending on nAddSize. May flush the previous blockfile to disk if full, updates
+     * blockfile info, and checks if there is enough disk space to save the block.
+     *
      * The nAddSize argument passed to this function should include not just the size of the serialized CBlock, but the also size of
      * separator fields which are written before it by WriteBlockToDisk (BLOCK_SERIALIZATION_HEADER_SIZE).
      */
-    [[nodiscard]] bool FindBlockPos(FlatFilePos& pos, unsigned int nAddSize, unsigned int nHeight, uint64_t nTime);
+    [[nodiscard]] FlatFilePos FindNextBlockPos(unsigned int nAddSize, unsigned int nHeight, uint64_t nTime);
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/29975#discussion_r1597601896

> does `[[nodiscard]]` still make sense now?

I think it does make sense, since the function can still fail and checking the return value is the only way to know.

Reply:
Agree. Not only for error-checking, we'd also expect users to do something with the `FlatFilePos` that `FindNextBlockPos` now returns.

Reply:
Would it still make sense to call this method and discard the return ed value, like before, when the output param was mutated?

Reply:
I don't think so, with the way the function is now only called if we don't know the BlockPos already it seems logical for callers to do something with the result. Of course, if someone comes up with such a use case in the future, they can just remove the `[[nodiscard]]` - after all its point is just to prevent future programming errors.